ui/src/components/tag/AddTagForm.js

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2021 Verizon Media
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import styled from '@emotion/styled';
+import InputLabel from '../denali/InputLabel';
+import Input from '../denali/Input';
+import { colors } from '../denali/styles';
+import Button from '../denali/Button';
+import Tag from '../denali/Tag';
+
+const SectionsDiv = styled.div`
+    width: 100%;
+    text-align: left;
+    background-color: ${colors.white};
+`;
+
+const SectionDiv = styled.div`
+    align-items: flex-start;
+    display: flex;
+    flex-flow: row nowrap;
+    padding: 10px 30px;
+`;
+
+const ContentDiv = styled.div`
+    flex: 1 1;
+    margin-right: 10px;
+`;
+
+const StyledInputLabel = styled(InputLabel)`
+    flex: 0 0 100px;
+    margin-right: 2%;
+`;
+
+const StyledInput = styled(Input)`
+    width: 500px;
+`;
+
+const ButtonDiv = styled.div`
+    margin-left: 10px;
+`;
+
+const StyledButton = styled(Button)`
+    width: 125px;
+`;
+
+const StyledIncludedValuesDiv = styled.div`
+    width: 65%;
+`;
+
+const StyledTag = styled(Tag)`
+    color: #d5d5d5;
+    &:hover {
+        background: #ffffff;
+    }
+    font-size: 14px;
+    height: 28px;
+    line-height: 14px;
+    margin: 5px 15px 5px 0;
+    padding: 6px 8px 7px 10px;
+`;
+
+const StyledTagColor = styled(StyledTag)`
+    background: rgba(53, 112, 244, 0.08);
+`;
+
+const StyledAnchor = styled.a`
+    text-decoration: none;
+`;
+
+const StyledAnchorActiveInline = { color: colors.linkActive };
+
+export default class AddTagForm extends React.Component {
+    constructor(props) {
+        super(props);
+        this.api = props.api;
+        this.addTagValue = this.addTagValue.bind(this);
+
+        if (props.editedTagKey && props.editedTagValues) {
+            this.state = {
+                newTagValue: '',
+                tagName: props.editedTagKey,
+                tagValues: props.editedTagValues,
+            };
+        } else {
+            this.state = {
+                newTagValue: '',
+                tagName: '',
+                tagValues: [],
+            };
+        }
+    }
+
+    inputChanged(key, evt) {
+        let value = '';
+        if (evt.target) {
+            value = evt.target.value;
+        } else {
+            value = evt ? evt : '';
+        }
+        this.setState({ [key]: value }, () =>
+            this.props.onUpdate(this.state.tagName, this.state.tagValues)
```

Comment:
we are not updating the tagValue on inputChanged because you are passing this.state.tagValue, whereas the input changed is this.state.newTagValue.

Reply:
This is relevant only for that:
`let tagNameChanged = this.inputChanged.bind(this, 'tagName');`
Anyway, deleted the inputChange for newTagValue. this is not relevant for parent component

Reply:
not sure I follow, 
the intended behavior in UI is:
user enters tag value in input field, but doesn't click on Add button. The user directly clicks on Submit button and it should work. This is similar to other add modals

Reply:
Exactly.
See in AddTag:
`        if (!this.state.tagValues || this.state.tagValues.length === 0) {
            this.setState({
                errorMessage: 'Tag value is required.',
            });
            return;
        }`

Reply:
so when the input field of tag value is changed then in addTagForm.js we do a setState on the varibale "newTagValue" but we call this.props.onUpdate with the variable "tagValues" which is empty until you click on Add Button

we don't wan't this

Reply:
In my first comment I wrote:
Anyway, deleted the inputChange for newTagValue. this is not relevant for parent component
So, we are updating the parent only on pressing Add button.

Reply:
Anyway, I think that we should not keep state of newTagValue